src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseParser.cs

```diff
@@ -187,7 +187,16 @@ public static unsafe bool GetResponseHeader(
                 index = originalIndex;
 
                 buffer = new char[bufferLength];
-                return GetResponseHeader(requestHandle, infoLevel, ref buffer, ref index, out headerValue);
+                fixed (char* pBuffer = &buffer[0])
+                {
+                    if (QueryHeaders(requestHandle, infoLevel, pBuffer, ref bufferLength, ref index))
+                    {
+                        headerValue = new string(pBuffer, 0, bufferLength);
+                        return true;
+                    }
+                }
+
+                int lastError = Marshal.GetLastWin32Error();
```

Comment:
I guess this should update the existing variable instead right?
```suggestion
                lastError = Marshal.GetLastWin32Error();
```

Reply:
Fixed